infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -260,7 +256,7 @@ private TxResolutionResponse txnCanCommit(TxResolutionInfo txInfo) {
             // for each key pair, check for conflict; if not present, check against the wildcard
             for (byte[] conflictParam : conflictParamSet) {
 
-                Long keyAddress = cache.getIfPresent(new ConflictTxStream(conflictStream.getKey(), conflictParam));
+                Long keyAddress = cache.getIfPresent(new ConflictTxStream(conflictStream.getKey(), conflictParam, Address.NON_ADDRESS));
```

Comment:
please use `long'

Reply:
So, the method by itself doesn't have any meaning and brings confusion. Please rename it to `get()` because the method is just a wrapper for the get method in a hashmap. 
Always, try to avoid having null as much as possible. In this case, it is pretty trivial to do.

I see several ways to go:
 1. use java.util.OptionalLong - the most convenient way in my opinion. It consumes the same amount of memory as Long but prevents you from null pointer exception. So, your method `getIfPresent` will return OptionalLong.
 2. use defaults, `long keyAddress = cache.getOrDefault(key, -1)` which also prevents you from NPE.
3. use https://bitbucket.org/trove4j/trove/src/master/ to avoid `autoboxing`. Since SequencerServer is a heavy loaded class, avoid using object when you keep only primitives dramatically improves performance and reduces memory consumption.

My opinion is to choose the first approach

 - Why you have to avoid using nulls https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/
 - autoboxing https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html

Reply:
Will not address.